SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp

```diff
@@ -196,3 +202,69 @@ void USpatialGameInstance::HandleOnConnectionFailed(const FString& Reason)
 #endif
 	OnConnectionFailed.Broadcast(Reason);
 }
+
+void USpatialGameInstance::OnLevelInitializedNetworkActors(ULevel* LoadedLevel, UWorld* OwningWorld)
+{
+	const FString WorkerType = GetSpatialWorkerType().ToString();
+
+	if (OwningWorld != GetWorld())
+	{
+		// Not current world. Another server running in the same process.
+		return;
+	}
+
+	if (!OwningWorld->IsServer())
+	{
+		return;
+	}
+
+	if (!GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
+	{
+		return;
+	}
+
+	if (OwningWorld->WorldType != EWorldType::PIE
+		&& OwningWorld->WorldType != EWorldType::Game
+		&& OwningWorld->WorldType != EWorldType::GamePreview)
+	{
+		return;
+	}
+
+	for (int32 ActorIndex = 0; ActorIndex < LoadedLevel->Actors.Num(); ActorIndex++)
+	{
+		AActor* Actor = LoadedLevel->Actors[ActorIndex];
+		if (Actor == nullptr)
+		{
+			continue;
+		}
+
+		if (USpatialStatics::IsSpatialOffloadingEnabled())
+		{
+			check(!GetDefault<USpatialGDKSettings>()->bEnableUnrealLoadBalancer);
+			if (!USpatialStatics::IsActorGroupOwnerForActor(Actor))
+			{
+				if (!Actor->bNetLoadOnNonAuthServer)
+				{
+					Actor->Destroy();
+				}
+				else
+				{
+					UE_LOG(LogSpatialGameInstance, Verbose, TEXT("WorkerType %s Not actor group owner of startup actor %s, Exchanging Roles"), *WorkerType, *GetPathNameSafe(Actor));
+					ENetRole Temp = Actor->Role;
+					Actor->Role = Actor->RemoteRole;
+					Actor->RemoteRole = Temp;
+				}
+			}
+		}
+
+		if (GetDefault<USpatialGDKSettings>()->bEnableUnrealLoadBalancer)
+		{
+			check(!USpatialStatics::IsSpatialOffloadingEnabled());
```

Comment:
I think we should prevent people from running with both enable rather than leaving this to runtime. Can we added something to `FSpatialGDKEditorToolbarModule::ValidateGeneratedLaunchConfig`?

Reply:
I see, yes, that would be a good place, but will probably leave the checks here as well, to catch cloud deployments and general other flows.

Reply:
Fair enough, can we check somewhere earlier though, i.e. the init method here?

Reply:
Moved to init method